src/testing/vortex/workload.zig

```diff
@@ -0,0 +1,432 @@
+        try reconcile(result, &command, &model);
+
+        const query = lookup_all_accounts(&model);
+        const query_result = try execute(query, driver) orelse break;
+        try reconcile(query_result, &query, &model);
+
+        log.info("accounts created = {d}, commands run = {d}", .{ model.accounts.items.len, i });
+    }
+}
+
+var accounts_buffer = std.mem.zeroes([accounts_count_max]tb.Account);
+
+const Command = union(enum) {
+    create_accounts: []tb.Account,
+    create_transfers: []tb.Transfer,
+    lookup_all_accounts: []u128,
+};
+
+const CommandBuffers = FixedSizeBuffers(Command);
+var command_buffers: CommandBuffers = std.mem.zeroes(CommandBuffers);
+
+const Result = union(enum) {
+    create_accounts: []tb.CreateAccountsResult,
+    create_transfers: []tb.CreateTransfersResult,
+    lookup_all_accounts: []tb.Account,
+};
+const ResultBuffers = FixedSizeBuffers(Result);
+var result_buffers: ResultBuffers = std.mem.zeroes(ResultBuffers);
+
+fn execute(command: Command, driver: *const DriverStdio) !?Result {
+    switch (command) {
+        .create_accounts => |events| return try execute_regular(
+            .create_accounts,
+            .create_accounts,
+            events,
+            driver,
+        ),
+        .create_transfers => |events| return try execute_regular(
+            .create_transfers,
+            .create_transfers,
+            events,
+            driver,
+        ),
+        .lookup_all_accounts => |events| return try execute_regular(
+            .lookup_all_accounts,
+            .lookup_accounts,
+            events,
+            driver,
+        ),
+    }
+}
+
+fn execute_regular(
+    comptime command_tag: std.meta.Tag(Command),
+    comptime operation: StateMachine.Operation,
+    events: []StateMachine.Event(operation),
+    driver: *const DriverStdio,
+) !?Result {
+    try send(driver, operation, events);
+
+    const buffer = @field(result_buffers, @tagName(command_tag))[0..events.len];
+    const results = receive(driver, operation, buffer) catch |err| {
+        switch (err) {
+            error.EndOfStream => return null,
+            else => return err,
+        }
+    };
+    return @unionInit(Result, @tagName(command_tag), results);
+}
+
+fn reconcile(result: Result, command: *const Command, model: *Model) !void {
+    switch (result) {
+        .create_accounts => |entries| {
+            const accounts_new = command.create_accounts;
+
+            // Track results for all new accounts, assuming `.ok` if response from driver is
+            // omitted.
+            var accounts_created: [accounts_count_max]struct {
+                tb.CreateAccountResult,
+                tb.Account,
+            } = undefined;
+            for (
+                accounts_new,
+                accounts_created[0..accounts_new.len],
+            ) |account_new, *account_created| {
+                account_created[0] = .ok;
+                account_created[1] = account_new;
```

Comment:
Feels like a useless copy here? Can you use accounts_new as is below? That is,
have two parallels arrays rather than an array of pairs?